pkg/apps/apis/apps/v1/types.go

```diff
@@ -33,12 +33,13 @@ type DeploymentConfig struct {
 	Spec DeploymentConfigSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
 
 	// Status represents the current deployment state.
-	Status DeploymentConfigStatus `json:"status" protobuf:"bytes,3,opt,name=status"`
+	Status DeploymentConfigStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
```

Comment:
doesn't do anything on a struct. does that matter?

Reply:
No, just using it as the canonical optional and "eventually someone will fix go json to not suck"